win32ss/user/user32/misc/exticon.c

```diff
@@ -439,35 +477,67 @@ static UINT ICO_ExtractIconExW(
                 DWORD dataOffset;
                 LPBYTE imageData;
                 POINT hotSpot;
+#ifndef __REACTOS__
                 LPICONIMAGE entry;
+#endif
 
                 dataOffset = get_best_icon_file_offset(peimage, fsizel, cx[index], cy[index], sig == 1, flags, sig == 1 ? NULL : &hotSpot);
 
                 if (dataOffset)
                 {
                     HICON icon;
                     WORD *cursorData = NULL;
+#ifdef __REACTOS__
+                    BITMAPINFOHEADER bmih;
+#endif
 
                     imageData = peimage + dataOffset;
+#ifdef __REACTOS__
+                    memcpy(&bmih, imageData, sizeof(BITMAPINFOHEADER));
+#else
                     entry = (LPICONIMAGE)(imageData);
+#endif
 
                     if(sig == 2)
                     {
+#ifdef __REACTOS__
+                         /* biSizeImage is the size of the raw bitmap data.
+                          * A dummy 0 can be given for BI_RGB bitmaps.
+                          * https://en.wikipedia.org/wiki/BMP_file_format */
+                         if ((bmih.biCompression == BI_RGB) &&
+                             (bmih.biSizeImage == 0))
+                         {
+                             bmih.biSizeImage = bmih.biWidth *
```

Comment:
Can you check `biClrUsed`?

Reply:
If `biClrUsed` is non-zero, there's a variable-length `RGBQUAD` color table...

Reply:
Check Added. Thanks.

Reply:
Sorry, my mistake. Please revert this.

Reply:
OK. Reverted. Thanks.